src/service/xhr-impl.js

```diff
@@ -344,17 +344,30 @@ function isRetriable(status) {
 /**
  * Returns the response if successful or otherwise throws an error.
  * @paran {!FetchResponse} response
- * @return {!FetchResponse}
+ * @return {!Promise<!FetchResponse|!Error>}
```

Comment:
@dvoytenko is this the accurate return type here to indicate resolve and reject responses?